### PR TITLE
add Exception module

### PR DESCRIPTION
**Pull request by gcanti**

### CODE REVIEW

File: Exception.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      return runEff(handler(e))
+    }
+  })
+}
+
+// Runs an Eff and returns eventual Exceptions as a `Left` value. If the
+// computation succeeds the result gets wrapped in a `Right`.
+export function tryEff<E: Object, A>(eff: Eff<E & { err: EXCEPTION }, A>): Eff<E, Either<Error, A>> {
+  return catchException((e) => inj(() => left(e)), map(right, eff))
+}
```
Comment (davidchambers) [inline Exception.js]: [`\n`][1] ;)


[1]: https://github.com/editorconfig/editorconfig/wiki/Newline-at-End-of-File-Support

Comment (gcanti) [inline Exception.js]: opss... thanks :)

Comment (davidchambers) [inline Exception.js]: No problem. :)

Comment (gcanti) [inline Exception.js]: @davidchambers if may I ask, what do you think about this POC?

Comment (davidchambers) [inline Exception.js]: I need to learn a lot about Flow and about this project before I'll be able to provide meaningful feedback, but I'm *very* excited to see that this sort of thing is possible!

**Pull request merged**